sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -187,6 +187,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val VERIFY_STATS_FROM_HDFS_WHEN_BROADCASTJOIN =
+    buildConf("spark.sql.statistics.verifyStatsFromHdfsWhenBroadcastJoin")
+    .doc("If table size in metastore is below spark.sql.autoBroadcastJoinThreshold, check the " +
+      "size on Hdfs and set table size to be the bigger one. This is for defense and help avoid" +
+      " OOM caused by broadcast join. It's useful when metastore failed to update the stats of" +
```

Comment:
Hi, @jinxing64 .
Is this helpful for highly compressed Parquet tables, too?

Reply:
@dongjoon-hyun It does not solve that issue.